packages/babel-parser/src/tokenizer/index.js

```diff
@@ -897,6 +897,19 @@ export default class Tokenizer extends LocationParser {
 
     let total = 0;
 
+    // Called from readHexChar
```

Comment:
Other than `readCodePoint`, `readHexChar` could also be called from `readRadixNumber`. Did you checked whether `0xf_f` can be successfully parsed?

We could add an extra `allowNumSeparator: boolean` to `readHexChar`
```js
readHexChar(len: number, throwOnInvalid: boolean, allowNumSeparator: boolean): number | null
```
and set `allowNumSeparator = true` when calling from `readCodePoint`.

Reply:
Good idea.